### PR TITLE
MULE-14152: Poll Doesn't wait for the execution to finish until it sends

### PR DESCRIPTION
**Pull request by elrodro83**
the next event

### CODE REVIEW

File: modules/spring-config/src/main/resources/META-INF/mule-core-common.xsd [hunk 1]
```diff
@@ -1493,6 +1493,13 @@
                 <xsd:sequence>
                     <xsd:element ref="scheduling-strategy" minOccurs="1" maxOccurs="1"/>
                 </xsd:sequence>
+                <xsd:attribute name="synchronous" default="false" type="xsd:string">
```
Comment (dfeist) [inline modules/spring-config/src/main/resources/META-INF/mule-core-common.xsd]: Can we use something clearer? `synchronous` makes it seem that it has to do with threading.  Are we sure best default behaviour is false?

File: modules/spring-config/src/main/resources/META-INF/mule-core-common.xsd [hunk 2]
```diff
@@ -1493,6 +1493,13 @@
...
+                    <xsd:annotation>
+                        <xsd:documentation>
+                            Whether additional polling triggers of the scheduler will be executed if there is another exceution currently running.
```
Comment (dfeist) [inline modules/spring-config/src/main/resources/META-INF/mule-core-common.xsd]: If `false` then will executions back-up? Should make this clear here and maybe reconsider the default..

File: modules/spring-config/src/main/resources/META-INF/mule-core-common.xsd (hunk 1 above)
Comment (elrodro83) [reply modules/spring-config/src/main/resources/META-INF/mule-core-common.xsd]: `false` as behavior is the current behaviour in 4.0.0, so we don't break compatibility. Let me think about the name.

File: modules/spring-config/src/main/resources/META-INF/mule-core-common.xsd (hunk 2 above)
Comment (elrodro83) [reply modules/spring-config/src/main/resources/META-INF/mule-core-common.xsd]: see previos response regarding the default

File: core/src/main/java/org/mule/runtime/core/internal/source/scheduler/DefaultSchedulerMessageSource.java [hunk 3]
```diff
@@ -125,21 +134,32 @@ private final void run() {
...
+    synchronized (this) {
+      if (disallowConcurrentExecution && executing) {
+        execute = false;
+      } else {
+        execute = true;
+        executing = true;
+      }
+    }
+
+    if (execute) {
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/internal/source/scheduler/DefaultSchedulerMessageSource.java]: Isn't it clearer to collapse this?

File: core/src/main/java/org/mule/runtime/core/internal/source/scheduler/DefaultSchedulerMessageSource.java [hunk 4]
```diff
@@ -125,21 +134,32 @@ private final void run() {
...
-
   /**
    * Triggers the forced execution of the polling message processor ignoring the configured scheduler.
    */
   private void poll() {
-    Message request = of(null);
-    pollWith(request);
+    boolean execute = false;
+    synchronized (this) {
+      if (disallowConcurrentExecution && executing) {
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/internal/source/scheduler/DefaultSchedulerMessageSource.java]: Maybe just style but i'd use:
 ```
if(executing.compareAndSet(false, true) || !disallowConcurrentExecution) {
  pollWith(of(null));
}
```

**Pull request merged**